projects/adrv9371x/common/adrv9371x_bd.tcl

```diff
@@ -135,45 +173,52 @@ ad_ip_parameter util_ad9371_xcvr CONFIG.QPLL_FBDIV 0x120
 
 # xcvr interfaces
 
-create_bd_port -dir I tx_ref_clk_0
-create_bd_port -dir I rx_ref_clk_0
-create_bd_port -dir I rx_ref_clk_2
+set tx_offset 0
+set rx_offset 0
+set rx_obs_offset  $RX_NUM_OF_LANES
 
-ad_xcvrpll  tx_ref_clk_0 util_ad9371_xcvr/qpll_ref_clk_0
-ad_xcvrpll  rx_ref_clk_0 util_ad9371_xcvr/cpll_ref_clk_0
-ad_xcvrpll  rx_ref_clk_0 util_ad9371_xcvr/cpll_ref_clk_1
-ad_xcvrpll  rx_ref_clk_2 util_ad9371_xcvr/cpll_ref_clk_2
-ad_xcvrpll  rx_ref_clk_2 util_ad9371_xcvr/cpll_ref_clk_3
-ad_xcvrpll  axi_ad9371_tx_xcvr/up_pll_rst util_ad9371_xcvr/up_qpll_rst_0
-ad_xcvrpll  axi_ad9371_rx_xcvr/up_pll_rst util_ad9371_xcvr/up_cpll_rst_0
-ad_xcvrpll  axi_ad9371_rx_xcvr/up_pll_rst util_ad9371_xcvr/up_cpll_rst_1
-ad_xcvrpll  axi_ad9371_rx_os_xcvr/up_pll_rst util_ad9371_xcvr/up_cpll_rst_2
-ad_xcvrpll  axi_ad9371_rx_os_xcvr/up_pll_rst util_ad9371_xcvr/up_cpll_rst_3
+set tx_ref_clk     tx_ref_clk_$tx_offset
+set rx_ref_clk     rx_ref_clk_$rx_offset
+set rx_obs_ref_clk rx_ref_clk_$rx_obs_offset
+
+create_bd_port -dir I $tx_ref_clk
+create_bd_port -dir I $rx_ref_clk
+create_bd_port -dir I $rx_obs_ref_clk
 ad_connect  sys_cpu_resetn util_ad9371_xcvr/up_rstn
 ad_connect  sys_cpu_clk util_ad9371_xcvr/up_clk
 
-ad_xcvrcon  util_ad9371_xcvr axi_ad9371_tx_xcvr axi_ad9371_tx_jesd
+# Tx
+ad_xcvrcon  util_ad9371_xcvr axi_ad9371_tx_xcvr axi_ad9371_tx_jesd {0 3 2 1}
```

Comment:
The order doesn't seem to be the same with the one before this commit. Was the TX tested to see if the channels haven't been swapped?

Reply:
Yes, they got swapped indeed, thanks for catching this.

Reply:
Link remapping is still needed, see the updated version. 
Previously it was done here : 
https://github.com/analogdevicesinc/hdl/blob/master/projects/adrv9371x/common/adrv9371x_bd.tcl#L163